### PR TITLE
util/metrics: collect threads state

### PR DESCRIPTION
**Pull request by overvenus**
Now `ThreadsColletcor` can monitor TiKV's threads state.

### CODE REVIEW

**Comment by breeswish** [comment]
What's the difference between the old one and the new one? Can we know more information than before?

**Comment by breeswish** [comment]
Oh forgive me. So.. in addition to thread cpu usage, we can collect thread state now?

**Comment by overvenus** [comment]
@breeswish Yes, besides CPU usage, it collects threads state.

**Comment by overvenus** [comment]
A friendly ping.

File: src/util/metrics/threads_linux.rs [hunk 1]
```diff
@@ -61,23 +69,41 @@ impl Collector for ThreadsCollector {
     }
 
     fn collect(&self) -> Vec<proto::MetricFamily> {
-        let cpu_totals = self.cpu_totals.lock().unwrap();
-        let tids = get_thread_ids(self.pid).unwrap();
+        // Synchronous collecting.
```
Comment (siddontang) [inline src/util/metrics/threads_linux.rs]: do we run the collect in multi-threads?

Comment (overvenus) [reply src/util/metrics/threads_linux.rs]: No, currently we collect metrics in a single-threading `promepusher`.

File: src/util/metrics/threads_linux.rs [hunk 2]
```diff
@@ -15,42 +15,50 @@ use std::fs;
...
-
-    Ok(())
+    prometheus::register(Box::new(tc)).map_err(|e| to_io_err(format!("{:?}", e)))
 }
 
+/// A collector to collect threads metrics, including CPU usage
+/// and threads state.
 struct ThreadsCollector {
-    pid: pid_t,
+    pid: Mutex<pid_t>,
```
Comment (huachaohuang) [inline src/util/metrics/threads_linux.rs]: Why do we need `Mutex` here?

Comment (huachaohuang) [reply src/util/metrics/threads_linux.rs]: I don't get it, what do you synchronize?

Comment (overvenus) [reply src/util/metrics/threads_linux.rs]: The `collect` method, to be precise, the CPU time, https://github.com/pingcap/tikv/pull/3173/files#diff-21ec373afb597cc6661bb2a2eb148e52R85

**Comment by breeswish** [comment]
LGTM

File: src/util/metrics/threads_linux.rs (hunk 2 above)
Comment (siddontang) [reply src/util/metrics/threads_linux.rs]: I am confused that why using Mutex for pid? seem it won't be changed.

Comment (overvenus) [reply src/util/metrics/threads_linux.rs]: Yes, it wont be changed. It's a lock for synchronous the `collect`. Of course, we can also choose other fileds.

**Comment by overvenus** [comment]
A friendly ping.

**Pull request merged**